Automattic/newspack-block-theme
# feat: add header desktop patterns

### All Submissions:






### Changes proposed in this Pull Request:

This PR uses the same concept as https://github.com/Automattic/newspack-block-theme/pull/177 but applies this to the Desktop Header.

I changed the template area to `header` so we can use the Template Part switcher(*) available in WP6.6 (see screencast)

We now have 5 generic patterns and we can add more in the future.

https://github.com/user-attachments/assets/4dd29440-4d79-4c2d-8cf1-a65d277e3745

(*) The downside is that now the "Header" template part (the one that contains Desktop && Mobile) can be switched but if a publisher does this we can always revert the template part...

### How to test the changes in this Pull Request:

1. Switch to this branch
2. Make sure you have WP6.6
3. Try editing the desktop header

### Other information:
